packages/fxa-auth-server/test/local/routes/subscriptions/apple.js

```diff
@@ -0,0 +1,177 @@
+const { default: Container } = require('typedi');
+const assert = { ...sinon.assert, ...require('chai').assert };
+const uuid = require('uuid');
+
+const mocks = require('../../../mocks');
+const {
+  AppleIapHandler,
+} = require('../../../../lib/routes/subscriptions/apple');
+const {
+  PurchaseUpdateError,
+} = require('../../../../lib/payments/iap/apple-app-store/types/errors');
+const error = require('../../../../lib/error');
+const { AuthLogger } = require('../../../../lib/types');
+const {
+  AppleIAP,
+} = require('../../../../lib/payments/iap/apple-app-store/apple-iap');
+const { IAPConfig } = require('../../../../lib/payments/iap/iap-config');
+const { OAUTH_SCOPE_SUBSCRIPTIONS_IAP } = require('fxa-shared/oauth/constants');
+const { CapabilityService } = require('../../../../lib/payments/capability');
+
+const MOCK_SCOPES = ['profile:email', OAUTH_SCOPE_SUBSCRIPTIONS_IAP];
+const ACCOUNT_LOCALE = 'en-US';
+const TEST_EMAIL = 'test@email.com';
+const UID = uuid.v4({}, Buffer.alloc(16)).toString('hex');
+const VALID_REQUEST = {
+  auth: {
+    credentials: {
+      scope: MOCK_SCOPES,
+      user: `uid1234`,
+      email: `test@testing.com`,
+    },
+  },
+};
+
+describe('AppleIapHandler', () => {
+  let iapConfig;
+  let appleIap;
+  let log;
+  let appleIapHandler;
+  let mockCapabilityService;
+  let db;
+
+  beforeEach(() => {
+    log = mocks.mockLog();
+    appleIap = {};
+    Container.set(AuthLogger, log);
+    iapConfig = {};
+    Container.set(IAPConfig, iapConfig);
+    Container.set(AppleIAP, appleIap);
+    db = mocks.mockDB({
+      uid: UID,
+      email: TEST_EMAIL,
+      locale: ACCOUNT_LOCALE,
+    });
+    db.account = sinon.fake.resolves({ primaryEmail: { email: TEST_EMAIL } });
+    mockCapabilityService = {};
+    mockCapabilityService.iapUpdate = sinon.fake.resolves({});
+    Container.set(CapabilityService, mockCapabilityService);
+    appleIapHandler = new AppleIapHandler(db);
+  });
+
+  afterEach(() => {
+    Container.reset();
+    sinon.restore();
+  });
+
+  describe('registerOriginalTransactionId', () => {
+    const request = {
+      ...VALID_REQUEST,
+      params: { appName: 'test' },
+      payload: { originalTransactionId: 'testTransactionId' },
+    };
+
+    it('returns valid with new products', async () => {
+      appleIap.purchaseManager = {
+        registerToUserAccount: sinon.fake.resolves({}),
+      };
+      iapConfig.getBundleId = sinon.fake.resolves('testPackage');
+      const result = await appleIapHandler.registerOriginalTransactionId(
+        request
+      );
+      assert.calledOnce(appleIap.purchaseManager.registerToUserAccount);
+      assert.calledOnce(iapConfig.getBundleId);
+      assert.calledOnce(mockCapabilityService.iapUpdate);
+      assert.deepEqual(result, { transactionIdValid: true });
+    });
+
+    it('returns valid with no changed products', async () => {
+      mockCapabilityService.subscribedProductIds = sinon.fake.resolves([
+        'prod_1234',
+      ]);
```

Comment:
This is no longer a method on the Capability Service, so we can remove this line from the test (and the corresponding test in test/local/routes/subscriptions/google.js).

Also, I'm not sure how this test case is supposed to differ from the one above it -- nothing else looks different. What is this supposed to be testing compared to the test above?

Reply:
Essentially its just to make sure that the function handles a no-change scenario the same as a change one. This is perhaps acting more as a test of the capability service, which isn't needed as the capability service test already handles this case.